File: 75 days Challenge Leetcode/Stack/Day-24/Solution.py
class Solution:
    def removeStars(self, s: str) -> str:
        result = []  # Use a list to collect characters for efficiency
        for c in s:  # Iterate through each character in the input string
            if c == '*':  # Check if the character is a star
                if result:  # If the result list is not empty
                    result.pop()  # Remove the last character from the result
            else:
                result.append(c)  # Add the current character to the result
        return ''.join(result)  # Join the list into a string and return

File: 75 days Challenge Leetcode/Stack/Day-24/Solution.cpp
#include <string>

class Solution {
public:
    string removeStars(string s) {
        string result;  
        for (char c : s) {  // Iterate through each character in the input string
            if (c == '*') {  // Check if the character is a star
                if (!result.empty())  // If the result string is not empty
                    result.pop_back();  // Remove the last character from the result
            } else 
                result.push_back(c);  // Add the current character to the result
        }
        return result;  // Return the final result
    }
};
